xclim/sdba/_adjustment.py

```diff
@@ -944,3 +944,378 @@ def extremes_adjust(
+            num_iter_max=num_iter_max,
+            jitter_inside_bins=jitter_inside_bins,
+        ),
+        input_core_dims=[["dim_hist", pts_dim], ["dim_ref", pts_dim]],
+        output_core_dims=[["dim_hist", pts_dim]],
+        keep_attrs=True,
+        vectorize=True,
+    )
+
+    # Pad dim differences with NA to please map_blocks
+    ref = ref.unstack().rename({v: k for k, v in ref_map.items()})
+    scen = scen.unstack().rename("scen")
+    for d in dim:
+        full_d = xr.concat([ref[d], scen[d]], dim=d).drop_duplicates(d)
+        scen = scen.reindex({d: full_d})
+
+    return scen.to_dataset()
+
+
+def _dotc_adjust(
+    X1: np.ndarray,
+    Y0: np.ndarray,
+    X0: np.ndarray,
+    bin_width: list | None = None,
+    bin_origin: list | None = None,
+    num_iter_max: int | None = 100_000_000,
+    cov_factor: str | None = "std",
+    jitter_inside_bins: bool = True,
+    kind: dict | None = None,
+):
+    """Dynamical Optimal Transport Correction of the bias of X with respect to Y.
+
+    Parameters
+    ----------
+    X1 : np.ndarray
+        Simulation data to adjust.
+    Y0 : np.ndarray
+        Bias correction reference.
+    X0 : np.ndarray
+        Historical simulation data.
+    bin_width : list | None
+        Bin widths for all dimensions.
+    bin_origin : list | None
+        Bin origins for all dimensions.
+    num_iter_max : int | None
+        Maximum number of iterations used in the earth mover distance algorithm.
+    cov_factor : str | None = "std"
+        Rescaling factor.
+
+    Returns
+    -------
+    np.ndarray
+        Adjusted data
+
+    References
+    ----------
+    :cite:cts:`sdba-robin_2021`
+    """
+    # Initialize parameters
+    if bin_width is None:
+        bin_width = u.bin_width_estimator([Y0, X0, X1])
+    elif isinstance(bin_width, list):
+        bin_width = np.array(bin_width)
+
+    # Map ref to hist
+    yX0 = _otc_adjust(
+        Y0,
+        X0,
+        bin_width=bin_width,
+        bin_origin=bin_origin,
+        num_iter_max=num_iter_max,
+        jitter_inside_bins=False,
+    )
+
+    # Map hist to sim
+    yX1 = _otc_adjust(
+        yX0,
+        X1,
+        bin_width=bin_width,
+        bin_origin=bin_origin,
+        num_iter_max=num_iter_max,
+        jitter_inside_bins=False,
+    )
+
+    # Temporal evolution
+    motion = np.empty(yX0.shape)
+    for j in range(yX0.shape[1]):
+        if kind is not None and j in kind.keys() and kind[j] == "*":
+            motion[:, j] = yX1[:, j] / yX0[:, j]
+        else:
+            motion[:, j] = yX1[:, j] - yX0[:, j]
+
+    # Apply a variance dependent rescaling factor
+    if cov_factor == "cholesky":
+        fact0 = u.eps_cholesky(np.cov(Y0, rowvar=False))
+        fact1 = u.eps_cholesky(np.cov(X0, rowvar=False))
+        motion = motion @ fact0 @ np.linalg.inv(fact1)
+    elif cov_factor == "std":
+        fact0 = np.std(Y0, axis=0)
+        fact1 = np.std(X0, axis=0)
+        motion = motion @ np.diag(fact0 / fact1)
```

Comment:
currently if `cov_factor` is not in {std, cholesky}, the algorithm silently applies no correction to the motion. 
1) If no correction to the motion is not intended, I think it should be an option (but it can just allow not be implemented since it's quite different from SBCK's implementation)
2) There should be an error if `cov_factor` is not in the proposed choices.
2)

Reply:
Is that check satisfying?
```
if cov_factor not in ["identity", "std", "cholesky"]:
    raise ValueError(
        "`cov_factor` should be in ['identity', 'std', 'cholesky']."
    )
```

Reply:
SBCK does have this option. Not sure of the use case, but since "cholesky" does not work with multiplicative kinds and "std" is univariate, maybe if something purely multivariate is needed?
Just thinking out loud, I'm open to remove the option if you think it would be useless.

Reply:
Also maybe the default should be "cholesky"? SBCK defaults to "std".

Reply:
I don't have a strong opinion on it, we can keep it in as `None` since it's also in SBCK. I think your choice of `std` as default is fine. What I like about this is that you can set `kind={"pr":"*"}` without an error by defaulting with `std`.  Perhaps you can mention the option cholesky in the notebook?